### PR TITLE
Do "lnet stop" instead of lustre_rmmod

### PR DESCRIPTION
**Pull request by utopiabound**
This should fix the hanging references from LU-11986

Signed-off-by: Nathaniel Clark <nclark@whamcloud.com>

### CODE REVIEW

**Comment by jgrund** [comment]
Testing this with 2.12 on https://build.whamcloud.com/job/integration-tests-shared-storage-configuration/1465

**Comment by jgrund** [comment]
Tested this locally, I'm seeing nodes freeze up during lnet config

File: chroma_agent/action_plugins/manage_lnet.py [hunk 1]
```diff
@@ -35,11 +35,10 @@ def start_lnet():
     """
     console_log.info("Starting LNet")
 
-    # modprobe lustre is a hack for HYD-1263 - Fix or work around LU-1279 - failure trying to mount
-    # should be removed when LU-1279 is fixed
     return agent_ok_or_error(
-        AgentShell.run_canned_error_message(["lctl", "net", "up"])
-        or AgentShell.run_canned_error_message(["modprobe", "lustre"])
+        # Do modprobe because stop does rmmod
+        AgentShell.run_canned_error_message(["modprobe", "lnet"])
```
Comment (jgrund) [inline chroma_agent/action_plugins/manage_lnet.py]: Shouldn't this be handled in `load_lnet `?

File: chroma_agent/action_plugins/manage_lnet.py [hunk 2]
```diff
@@ -76,7 +75,11 @@ def unload_lnet():
 
     Lnet must be stopped before unload_lnet is called.
     """
-    return agent_ok_or_error(AgentShell.run_canned_error_message(["lustre_rmmod"]))
+    if not os.path.exists("/dev/lnet"):
+        return agent_ok
```
Comment (jgrund) [inline chroma_agent/action_plugins/manage_lnet.py]:
```suggestion
        return agent_result_ok
```

**Comment by jgrund** [comment]
> Can you test with latest? It should be working now.

Testing with b2_12 now

File: chroma_agent/action_plugins/manage_lnet.py (hunk 1 above)
Comment (utopiabound) [reply chroma_agent/action_plugins/manage_lnet.py]: Yes, but this is working around LU-11986.

Comment (jgrund) [reply chroma_agent/action_plugins/manage_lnet.py]: Ok, I am testing with LU-11986, so if you remove this I'll push to Copr and test again.

File: chroma_agent/action_plugins/manage_lnet.py [hunk 3]
```diff
@@ -54,7 +53,7 @@ def stop_lnet():
     # network down" won't work...) call lustre_rmmod to shutdown network
 
     # Teardown to ksocklnd - ignore error
-    result = AgentShell.run(["lustre_rmmod"])
+    result = AgentShell.run(["/usr/sbin/lnet", "stop"])
```
Comment (jgrund) [inline chroma_agent/action_plugins/manage_lnet.py]: Unless I'm missing something, `/usr/sbin/lnet` does not appear to be provided in the lustre client RPM.

Comment (utopiabound) [reply chroma_agent/action_plugins/manage_lnet.py]: Is it needed on the client?

Comment (jgrund) [reply chroma_agent/action_plugins/manage_lnet.py]: yes, clients have access to lnet actions

File: chroma_agent/action_plugins/manage_lnet.py [hunk 4]
```diff
@@ -53,11 +53,10 @@ def stop_lnet():
...
 
     # Teardown to ksocklnd - ignore error
-    result = AgentShell.run(["/usr/sbin/lnet", "stop"])
-    if result.rc == 0:
-        return agent_result_ok
     return agent_ok_or_error(
-        AgentShell.run_canned_error_message(["lctl", "network", "down"])
+        AgentShell.run_canned_error_message(["lustre_rmmod", "ptlrpc"])
+        or AgentShell.run_canned_error_message(["lnetctl", "lnet", "unconfigure"])
+        or AgentShell.run_canned_error_message(["lctl", "network", "down"])
```
Comment (jgrund) [inline chroma_agent/action_plugins/manage_lnet.py]: Is this necessary?

Comment (utopiabound) [reply chroma_agent/action_plugins/manage_lnet.py]: I'm not sure the `lctl net down` is actually necessary, but even if lnet is already down, it returns 0.

**Pull request merged**